mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp

```diff
@@ -713,6 +718,8 @@ struct WarpOpConstant : public OpRewritePattern<WarpExecuteOnLane0Op> {
     auto dense = dyn_cast<SplatElementsAttr>(constantOp.getValue());
     if (!dense)
       return failure();
+    // Notify the rewriter that the warp op is changing.
+    rewriter.startRootUpdate(warpOp);
```

Comment:
Later, would be nice to wrap it with `make_scope_exit(...)`, to account for all the exits, but that can be a cleanup PR.

Reply:
makes sense, can do that later.